### ISSUE TITLE
Restore existing behavior when rendering document title (<title> tag)

### ISSUE THREAD

**Issue by danielbachhuber** [WordPress/gutenberg#3702]
I've noticed that Gutenberg edits the `<title>` tag:

![image](https://user-images.githubusercontent.com/36432/33339978-99211f02-d42f-11e7-9993-9d3d2a9089af.png)

As an existing WordPress writer, this makes it difficult to find the tab I was writing in because I'm accustomed to searching for "Edit Post":

![image](https://user-images.githubusercontent.com/36432/33340041-cb5c3b96-d42f-11e7-8381-1334dd62a383.png)

Just wanted to record this usability feedback.

**Comment by afercia** [comment]
This is important for accessibility too, as the document title is the first thing screen readers announce. Not sure what the reasoning behind this is, but at least the initial document title should be consistent with what WordPress does in the Classic Editor and say as first thing _what_ the page is about:
Edit Post ...
Edit Page ...
Edit {CPT name here} ...

**Comment by mtias** [comment]
It was added here: https://github.com/WordPress/gutenberg/pull/1802

> Only other thought I have is: Do we really want the document title to be assigned as the post's title? Per #1737 and current editor behavior, seems "Edit Post" could work just as well. Was there any conversation or thoughts behind the choice to use post title?

cc @westonruter @aduth

**Comment by westonruter** [comment]
If you have multiple posts open for editing in multiple tabs, having the post's title in the document title is key to be able to identify which tab contains the post you're needing to get back to editing. The post title was put at the beginning to account for browser tabs that are narrow, as in screenshot. But prefixing the title with “Edit…” would probably be a good compromise. (Or else, a ✏️ emoji perhaps? 😄)

**Comment by afercia** [comment]
> (Or else, a ✏️ emoji perhaps? 😄) 

lol. Well I understand the argumentation, however Gutenberg shouldn't deviate from what WordPress does in all the other screens. The document title should be consistent across all admin screens. Not sure adding the post title would fit in the limited space (browsers tend to truncate long titles); we should also consider the doc title has a limited _visual_ value, while it is very important for other software reading (and announcing) it.

**Comment by karmatosed** [comment]
I'm not sure we're being serious but a -1 on emoji's here, they are awesome but not in this case :)

I think it's right to go with existing WP patterns in this case. We can always iterate on them later.

**Comment by fatcat32594** [comment]
I've done a very small edit which adds an 'Edit: ' to the beginning of the editor's <title> tag, like so, but which otherwise keeps the post title too.
![new-editor-title-example](https://user-images.githubusercontent.com/4240738/36300723-4a9a7fc4-12d0-11e8-8356-8cf869409d1a.png)

**Comment by danielbachhuber** [comment]
To better document this issue...

Gutenberg does a pretty mediocre job of rendering the document title `<title>`.

Specifically, when creating a new post, Gutenberg shoves the title in front of `Add New Post ‹`:

<img width="293" alt="image" src="https://user-images.githubusercontent.com/36432/40544264-88969d90-5fdc-11e8-8930-8c0b3a0de918.png">

Then, once you've saved the post and refreshed the page, Gutenberg ends up in this state:

<img width="280" alt="image" src="https://user-images.githubusercontent.com/36432/40544305-aa8d1a5a-5fdc-11e8-9244-4ef98a7ceb7a.png">

Because the document title can be both localized _and_ filtered with `admin_title`, I think we should ditch this client-side modification and only set `<title>` server-side.

**Comment by danielbachhuber** [comment]
> Currently this is more-or-less true.

The current implementation is broken though?

> Without client-side modification as proposed, however, I expect this will no longer be true.

If we switch back to purely server-side, then the `<title>` tag should remain consistent regardless of state.

**Comment by danielbachhuber** [comment]
To avoid the complexity of having to deal with two separate styles of document `<title>` (one for `post-new.php` and a second for `post.php?post=<id>`), maybe we could _always_ redirect `post-new.php` to `post.php?post=<id>` and avoid `post-new.php` entirely.

I'm unsure of the history for `post-new.php` / `post.php?post=<id>`, so we'd need to vet this from a back-compat perspective. I'm not aware of any immediate back-compat concerns though.

**Comment by pento** [comment]
Always redirecting `post-now.php` to `post.php` has the potential for a whole host of subtle bugs that rely on the `WP_Screen` functionality, or the `$parent_file` / `$submenu_file` values. I would like to avoid it if possible.

Given that the `<title>` doesn't include the `post_title`, I think it's reasonable to duplicate the logic in `admin-header.php` to generate the "Edit Post" `<title>`, and swap that in when the new post is saved.

**Comment by danielbachhuber** [comment]
While I'm thinking of it, the other thing that needs to dynamically update is this:

<img width="491" alt="image" src="https://user-images.githubusercontent.com/36432/40982879-3ca1b66c-6893-11e8-8491-e1ac877fed6d.png">

> Always redirecting `post-now.php` to `post.php` has the potential for a whole host of subtle bugs that rely on the `WP_Screen` functionality, or the `$parent_file` / `$submenu_file` values. I would like to avoid it if possible.

Arguably, these bugs are already present in Gutenberg, given the editing experience starts at `post-new.php` and doesn't load `post.php` until page refresh, whereas the Classic editor opens `post.php` after the first save.

While I don't have a strong opinion about the final implementation, I'm not sure the information you've presented is an argument in favor of your suggested implementation. I think always redirecting is an option still worth considering.

**Comment by designsimply** [comment]
I removed the `Good First Issue` label and added the `Needs Decision` label because the following should be addressed before this issue is actionable:

1. Should `<title>` be prefixed with `Edit:` or `Edit Post ‹ ` while editing a post with Gutenberg?
1. Should `<title>` not be changed client-side because it can be localized and filtered with `admin_title`?
1. Should we always redirect post-new.php to post.php?post=<id> (avoid post-new.php entirely) so we don't have to deal with two separate styles of `<title>`?

**Comment by danielbachhuber** [comment]
At this point, my vote would be:

1. Gutenberg always displays `Edit Post ‹ Site Title — WordPress`
2. `<title>` is not changed client-side.
3. `<title>` can be filtered by the `admin_title` filter.
4. Gutenberg does not redirect `post-new.php` to `post.php?post=`, and maintains the same `<title>` between them.

I think this is the simplest implementation that maintains the closest parity to Classic Editor behavior.

**Comment by aduth** [comment]
>Gutenberg does not redirect `post-new.php` to `post.php?post=`

I'm not really so fond of this one, largely because I think it's a common pattern to reload the page (as an explicit user interaction, or maybe just reopening closed tabs). And if I'd started and saved a post, I'd expect a reload to bring me to that post; not a blank new post.

Would it be enough to just remove client-side title manipulations for now? I don't know that they're adding much value as it is.

**Comment by danielbachhuber** [comment]
> I'm not really so fond of this one, largely because I think it's a common pattern to reload the page (as an explicit user interaction, or maybe just reopening closed tabs). And if I'd started and saved a post, I'd expect a reload to bring me to that post; not a blank new post.

Ah, I see that you're referring to Gutenberg's existing behavior of updating the URL after initial save. I think this behavior should persist. I was referring to forcefully redirecting `post-new.php` on PHP page load.

I think we should make these changes:

1. Remove client-side manipulation of `<title>`.
2. Always default `<title>` to `Edit Post ‹ Site Title — WordPress`, regardless of whether the page is `post-new.php` or `post.php?post=`.

I can prepare a PR
